coreservices/partsrelationshipservice/prs-api/src/main/java/net/catenax/prs/services/PartsTreeQueryByVinService.java

```
@@ -62,19 +64,19 @@
                         .value(PrsConfiguration.VEHICLE_ATTRIBUTE_VALUE).build());
         final var vehicles = attributeRepository.findAll(searchFilter, SORTED_BY_ONEID);
         if (vehicles.isEmpty()) {
-            return Optional.empty();
+            throw new EntityNotFoundException(MessageFormat.format(ApiErrors.VEHICLE_NOT_FOUND_BY_VIN, request.getVin()));
```

Why are we throwing an exception here instead of returning an empty list of relationships?

---

So behavior remains same with this change, even before we were returning 404 when VIN not found. Added this exception so we can response with a error body explaining details of error.

Also with OneId & OneObjectId we returns empty list of relationships because its possible that pair exist as children but not as parent that's why we respond back with empty list.  But VIN we say not found as its a single identifier which only act as root of the tree.

But overall with this changes behavior of API doesn't change it is just for adding error response.

---

ok, thanks for explanation